enhancements/sig-architecture/225-clusteradm-operator/README.md

```diff
@@ -0,0 +1,557 @@
+		return fmt.Sprintf("cpu=%s", r.CPU)
+	} else if r.Memory != "" {
+		return fmt.Sprintf("memory=%s", r.Memory)
+	}
+	return ""
+}
+
+// RegistrationAuth provides specifications for registration authentication.
+type RegistrationAuth struct {
+	// The registration authentication driver to use.
+	// Options are:
+	//  - csr: Use the default CSR-based registration authentication.
+	//  - awsirsa: Use AWS IAM Role for Service Accounts (IRSA) registration authentication.
+	// The set of valid options is open for extension.
+	// +kubebuilder:validation:Enum=csr;awsirsa
+	// +kubebuilder:default:="csr"
+	Driver string `json:"driver"`
+
+	// The Hub cluster ARN for awsirsa registration authentication. Required when Type is awsirsa, otherwise ignored.
+	HubClusterARN string `json:"hubClusterARN,omitempty"`
+
+	// List of AWS EKS ARN patterns so any EKS clusters with these patterns will be auto accepted to join with hub cluster.
+	// Example pattern: "arn:aws:eks:us-west-2:123456789013:cluster/.*"
+	AutoApprovedARNPatterns []string `json:"autoApprovedARNPatterns,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="PHASE",type=string,JSONPath=`.status.phase`
+// +kubebuilder:printcolumn:name="AGE",type=date,JSONPath=".metadata.creationTimestamp"
+
+// MultiCluster is the Schema for the multiclusters API.
+type MultiCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MultiClusterSpec   `json:"spec,omitempty"`
+	Status MultiClusterStatus `json:"status,omitempty"`
+}
+
+// GetDriver returns the registration auth type, defaults to csr.
+func (ra *RegistrationAuth) GetDriver() string {
+	if ra == nil {
+		// default registration auth type
+		return CSRRegistrationDriver
+	}
+	return ra.Driver
+}
+
+// GetCondition gets the condition with the supplied type, if it exists.
+func (m *MultiCluster) GetCondition(cType string) *Condition {
+	return m.Status.GetCondition(cType)
+}
+
+// SetConditions sets the supplied conditions on a MultiCluster, replacing any existing conditions.
+func (m *MultiCluster) SetConditions(cover bool, c ...Condition) {
+	m.Status.SetConditions(cover, c...)
+}
+
+// +kubebuilder:object:root=true
+
+// MultiClusterList contains a list of MultiCluster.
+type MultiClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []MultiCluster `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&MultiCluster{}, &MultiClusterList{})
+}
+```
+
+### Test Plan
+
+The following scenarios will be covered via E2E tests:
+
+- Installation
+  - Hub-as-spoke
+  - Dedicated hub + spoke (IaaS)
+  - Dedicated hub + spoke (EKS)
+- Patches/Upgrades
+  - Feature gates (not actually supported at this time, requires feature implementation in `clusteradm` first)
+  - Helm values (singleton)
+  - Upgrade OCM bundle version
+- Uninstallation
+
+### Graduation Criteria
+
+TBD
```

Comment:
I think we also iterate on some alternatives. For example:

User can set a global or per cluster configs, so when the user create a managedcluster, the agent will be deployed to the target cluster automatically.

Reply:
Qi is referring to the ACM pattern for importing an existing cluster. There's an [import controller](https://github.com/stolostron/managedcluster-import-controller) that watches for the creation of a ManagedCluster, along with its namespace and a way to connect to the spoke cluster (like a kubeconfig Secret). Once those are in place, it can connect to the managed cluster, deploy the klusterlet operator, and create the Klusterlet CR to initiate the registration process.

Reply:
I mean you will still need to start "clusteradm operator" on the hub/spoke, and still need to handle the version and lifecycle of this operator. It is not clear to me why user cannot just install ocm operators directly. It would need some clarification.

Reply:
Everything you said is true. The value add that the proposed operator provides is around consolidation and minimizing manual steps. Users would have a single cluster with network connectivity to all other clusters that acts similarly to a CAPI management cluster. Say you have 20 clusters that you want to link together. With this approach you would only need to install the multi-cluster operator on one cluster and apply a single CR.

Reply:
the goal is very similar as what we did on capi-importer work https://github.com/open-cluster-management-io/ocm/tree/main/pkg/registration/hub/importer. The difference is it happens the join happens when a managedcluster is created on the hub. Would you also take a look at to see if the existing approach is good enough and can be extended to support your goal?